### PR TITLE
Update ansible role so that it supports adding images from image_url

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -28,6 +28,13 @@
     state: directory
   with_items: "{{ os_images_list }}"
 
+- name: Download the image to the directory when an image_url is defined
+  get_url:
+    url: "{{ item.image_url }}"
+    dest: "{{ os_images_cache }}/{{ item.name }}/{{ item.name }}.qcow2"
+  with_items: "{{ os_images_list }}"
+  when: item.image_url is defined
```
Comment (oneswig) [inline tasks/main.yml]: How about registering the download result, and (in a similar manner to the builds, below) add a check to fail if any downloads were not successful.

**Review by oneswig — COMMENTED** [review]
small thing to add.

File: tasks/main.yml [hunk 2]
```diff
@@ -156,8 +170,8 @@
         disk_format: "{{ item.type | default('qcow2') }}"
         filename: "{{ os_images_cache }}/{{ item.0.name }}/{{ item.0.name }}.{{ item.type | default('qcow2') }}"
         properties: "{{ os_images_common_properties | combine(item.0.properties | default({})) or omit }}"
-        kernel: "{{ item.1.id if 'baremetal' in item.0.elements else omit }}"
-        ramdisk: "{{ item.2.id if 'baremetal' in item.0.elements else omit }}"
+        kernel: "{{ item.1.id if item.0.elements is defined and 'baremetal' in item.0.elements else omit }}"
+        ramdisk: "{{ item.2.id if item.0.elements is defined and 'baremetal' in item.0.elements else omit }}"
```
Comment (markgoddard) [inline tasks/main.yml]: Protip: When variable expressions get busy like this, you can use task-local variables to clean things up.

```
  ramdisk: "{{ item.2.id if is_baremetal else omit }}"
vars:
  is_baremetal: "{{ item.0.elements is defined and 'baremetal' in item.0.elements }}"
```

File: tasks/main.yml [hunk 3]
```diff
@@ -28,6 +28,13 @@
     state: directory
   with_items: "{{ os_images_list }}"
 
+- name: Download the image to the directory when an image_url is defined
+  get_url:
+    url: "{{ item.image_url }}"
+    dest: "{{ os_images_cache }}/{{ item.name }}/{{ item.name }}.qcow2"
```
Comment (markgoddard) [inline tasks/main.yml]: We can support multiple image types if we `s/qcow2/{{ item.type | default('qcow2') }}/`.

**Review by markgoddard — CHANGES_REQUESTED** [review]
This looks sensible. Can you update the description of os_images_list in defaults/main.yml and the README?

File: tasks/main.yml (hunk 1 above)
Comment (brtkwr) [reply tasks/main.yml]: I just tested this with a dummy URL and looks like Ansible automatically fails if the URL is unreachable.

File: defaults/main.yml [hunk 4]
```diff
@@ -25,6 +25,11 @@ os_images_common: cloud-init enable-serial-console stable-interface-names
 #     env:
 #       DIB_XXX: yyy
 #     type: qcow2
+#   - name: FedoraAtomic27
+#     image_url: https://ftp.icm.edu.pl/pub/Linux/dist/fedora-alt/atomic/stable/Fedora-Atomic-27-20180326.1/CloudImages/x86_64/images/Fedora-Atomic-27-20180326.1.x86_64.qcow2
+#     properties:
+#       os_distro: fedora-atomic
+#     type: qcow2
```
Comment (markgoddard) [inline defaults/main.yml:32]: This is good, but also needs documenting in the readme.

**Review by markgoddard — APPROVED** [review]
Looking good, thanks.

**Pull request merged**